auto_submit/lib/validations/ci_successful.dart

```diff
@@ -193,7 +208,21 @@ class CiSuccessful extends Validation {
       }
       allSuccess = false;
     }
+    if (staleCheckRuns.isNotEmpty) {
+      log.warning(
+        'Pull request https://github.com/${slug.fullName}/pull/$prNumber from ${slug.name} repo auto roller has been running over ${Config.kGitHubCheckStaleThreshold} hours due to: ${staleCheckRuns.map((e) => e.name).toList()}',
+      );
+    }
 
     return allSuccess;
   }
+
+  // Treat any GitHub check run as stale if created over [Config.kGitHubCheckStaleThreshold] hours ago.
+  bool isStale(DateTime dateTime) {
+    return dateTime.compareTo(DateTime.now().subtract(const Duration(hours: Config.kGitHubCheckStaleThreshold))) < 0;
+  }
+
+  bool isEngineRoller(Author author, github.RepositorySlug slug) {
+    return config.rollerAccounts.contains(author.login!) && slug == Config.engineSlug;
+  }
```

Comment:
@zanderso My earlier change is for rolled PRs landing in Engine repo. I guess we also want to include the engine to framework ones?

Reply:
Yes.